### PR TITLE
Remove old functions from the CCDB api

### PR DESCRIPTION
**Pull request by Barthelemy**

### CODE REVIEW

File: CCDB/include/CCDB/CcdbApi.h [hunk 1]
```diff
@@ -97,17 +97,6 @@ class CcdbApi //: public DatabaseInterface
...
-   * Retrieve object at the given path for the given timestamp.
-   *
-   * @param path The path where the object is to be found.
-   * @param metadata Key-values representing the metadata to filter out objects.
-   * @param timestamp Timestamp of the object to retrieve. If omitted, current timestamp is used.
-   * @return the object, or nullptr if none were found.
-   */
-  TObject* retrieve(std::string const& path, std::map<std::string, std::string> const& metadata,
-                    long timestamp = -1) const;
-
```
Comment (sawenzel) [inline CCDB/include/CCDB/CcdbApi.h]: shouldn't you also take away the declaration of store, in addition?

**Comment by sawenzel** [comment]
Some people are still using store/retrieve. I will fix this and amend to this PR.

**Comment by sawenzel** [comment]
@Barthelemy : We are almost there. I think you still need to adjust code in QualityControl:
```
/mnt/mesos/sandbox/sandbox/sw/SOURCES/QualityControl/v0.17.0/v0.17.0/Framework/src/CcdbDatabase.cxx: In member function 'virtual void o2::quality_control::repository::CcdbDatabase::store(std::shared_ptr<o2::quality_control::core::MonitorObject>)':
/mnt/mesos/sandbox/sandbox/sw/SOURCES/QualityControl/v0.17.0/v0.17.0/Framework/src/CcdbDatabase.cxx:114:11: error: 'class o2::ccdb::CcdbApi' has no member named 'store'
   ccdbApi.store(mo.get(), path, metadata, from, to);
           ^~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/QualityControl/v0.17.0/v0.17.0/Framework/src/CcdbDatabase.cxx: In member function 'virtual o2::quality_control::core::MonitorObject* o2::quality_control::repository::CcdbDatabase::retrieve(std::__cxx11::string, std::__cxx11::string, long int)':
/mnt/mesos/sandbox/sandbox/sw/SOURCES/QualityControl/v0.17.0/v0.17.0/Framework/src/CcdbDatabase.cxx:127:22: error: 'class o2::ccdb::CcdbApi' has no member named 'retrieve'; did you mean 'retrieveBlob'?
     object = ccdbApi.retrieve(path, metadata, when);
                      ^~~~~~~~
                      retrieveBlob
```

Otherwise it is looking quite good and we also have the unit test running on MacOS.

**Comment by Barthelemy** [comment]
The code is already there but I need to release 0.18.

**Comment by Barthelemy** [comment]
ok my bad, there is still an issue in QC. Fixing it.

**Comment by Barthelemy** [comment]
Now that I look into it more carefully, I am afraid that pushing this change will prevent some users from reading their old data as the `retrieve` is not available any more.
@sawenzel Would you be ok that I amend this PR by leaving `retrieve` in the API but remove all the rest ? it is not perfect but it is a step forward and avoids hurting users.

**Comment by sawenzel** [comment]
Ok for me.

**Comment by sawenzel** [comment]
Ok for me. Waiting for approval from @mfasDa since this is touching EMCAL code.

**Comment by Barthelemy** [comment]
ok, happy to see that it is finally green

**Comment by sawenzel** [comment]
Squash-merging. @mfasDa approved via private communication.

**Pull request merged**